dateutil/parser/_parser.py

```diff
@@ -571,6 +571,10 @@ def resolve_ymd(self, yearfirst, dayfirst):
 
 
 class parser(object):
+    # timelex_cls is a class attribute so it can be easily overriden in
+    # subclasses
+    _timelex_cls = _timelex
```

Comment:
Both `_timelex_cls` and `_timelex` are private attributes. I think this is the right thing to do in the end, but in the short term the only effect this will have is encouraging people to make use of the private interface.

Reply:
I'm not clear on what you're suggesting?  Are you saying `_timelex_cls` should be public?

I think it makes sense to keep private attributes that end-users don't need to worry about (and/or shouldn't mess with).  If someone is sub-classing, it's less obvious how meaningful the public/private distinction is.

Reply:
I'm saying this change is not necessary because we don't need the `_timelex_cls` attribute internally (since we have exactly one lexer) and no one who is *not* `dateutil` should be writing code that relies on the private interface (whcih are assumed to be implementation details).

Whether or not someone is subclassing does not really change the meaning of public vs. private. From the perspective of someone subclassing your class, you should assume that *only* the public parts of the interface exist (other than trying to avoid namespace collisions). Everything else is an implementation detail.

For example, we may write a drop-in replacement for the parser with a compiled backend, in which case likely we would not write Python versions of the private interface, because that would just slow things down and we've already advertised that no one should be relying on the details of the implementation by marking those things private. Any code relying on the existence of that private interface would break when it encounters the compiled version.

Reply:
I'm having trouble squaring what you're saying here with the opinions expressed in #123 and #139.  Has something changed your mind?

Let's avoid delving into the semantics of "public" vs "private" vs "implementation detail".

Reply:
I would like to eventually make the lexer public, but it is not currently public.

Adding a private attribute to `parser` that we, the only consumers of the private interface, do not need, does not help anything. When `timelex` or equivalent is made public, depending on the details we decide for the API, we may add a *public* attribute like `lexer_class`, but this private attribute would have to be replaced when that happens anyway, so it doens't help anything in the long or short run.

Reply:
> but this private attribute would have to be replaced when that happens anyway,

We can worry about that when we get there.  It's not as if this is on the immediate horizon.

> so it doens't help anything in the long or short run.

de facto pandas uses _timelex, and would benefit from being able to drop in a simplified+cythonized version of it.

Plus making it available to early adapters will get some feedback and make design decisions for an eventual public version better-informed.

Reply:
>de facto pandas uses _timelex, and would benefit from being able to drop in a simplified+cythonized version of it.

I have said many times that pandas should not be using `_timelex`. I will likely *not* continue to go through deprecation cycles for very explicitly private code in the future as I have last time this came up. Enabling people to do stuff like this is exactly why this kind of thing has caused problems in the past.

> Plus making it available to early adapters will get some feedback and make design decisions for an eventual public version better-informed.

At this point there's nothing to give feedback on.